src/api/apiMethods.js

```diff
@@ -55,54 +56,162 @@ const getApiMethodsSelector = () =>
         scrollableColumnGroups,
       } = columnWidths;
 
+      const _getCellGroupTypeFromColumnIndex = (columnIndex) => {
+        const totalColumns =
+          fixedColumns.length +
+          scrollableColumns.length +
+          fixedRightColumns.length;
+
+        if (
+          columnIndex < 0 ||
+          columnIndex > totalColumns ||
+          !Number.isInteger(columnIndex)
+        ) {
+          throw `columnIndex must be an integer between 0 and ${
+            totalColumns - 1
+          } inclusive`;
+        }
+
+        if (columnIndex < fixedColumns.length) {
+          return CellGroupType.FIXED;
+        }
+
+        if (columnIndex < fixedColumns.length + scrollableColumns.length) {
+          return CellGroupType.SCROLLABLE;
+        }
+
+        return CellGroupType.FIXED_RIGHT;
+      };
+
+      const _getCellGroupTypeFromColumnGroupIndex = (columnGroupIndex) => {
+        const totalColumnGroups =
+          fixedColumnGroups.length +
+          scrollableColumnGroups.length +
+          fixedRightColumnGroups.length;
+
+        if (
+          columnGroupIndex < 0 ||
+          columnGroupIndex > totalColumnGroups ||
+          !Number.isInteger(columnGroupIndex)
+        ) {
+          throw `columnGroupIndex must be an integer between 0 and ${
+            totalColumnGroups - 1
+          } inclusive`;
+        }
+
+        if (columnGroupIndex < fixedColumnGroups.length) {
+          return CellGroupType.FIXED;
+        }
+
+        if (
+          columnGroupIndex <
+          fixedColumnGroups.length + scrollableColumnGroups.length
+        ) {
+          return CellGroupType.SCROLLABLE;
+        }
+
+        return CellGroupType.FIXED_RIGHT;
+      };
+
+      const _getColumnLocalIndex = (columnIndex, cellGroupType) => {
```

Comment:
```suggestion
      const _getLocalColumnIndex = (columnIndex, cellGroupType) => {
```

Reply:
To me suggested name is more confusing, it sounds like index of some "local column" where it is "local index" of some column